src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -653,7 +676,9 @@ AmendmentTableImpl::doVoting(
             JLOG(j_.debug()) << entry.first << ": amendment got majority";
             actions[entry.first] = tfGotMajority;
         }
-        else if (!hasValMajority && (majorityTime != NetClock::time_point{}))
+        else if (
+            !hasValMajority && (majorityTime != NetClock::time_point{}) &&
+            !vote->passes(entry.first, vote->deactivationThreshold()))
```

Comment:
I _think_ this is right and should have the same behavior as the existing code, but someone else should triple-check the logic when the amendment being checked is unknown. Here's the logic: `hasValMajority` will be `false`; before this commit, we'd always enter this block. Now `vote->passes` will return `false`, and the `!` will turn it to `true` putting us in this block, just like before.

Reply:
If the amendment is unknown, then how would `majorityTime` be non-zero? And if it is zero for unknown amendments, then how would this branch have been entered before the change?